[stdlib] Adopt `@_inheritActorContext(always)` on `Task.immediate`

xedin opened the pull request:
This change aligns implementation with the current SE-0472 proposal.

<!--
If this pull request is targeting a release branch, please fill out the
following form:
https://github.com/swiftlang/.github/blob/main/PULL_REQUEST_TEMPLATE/release.md?plain=1

Otherwise, replace this comment with a description of your changes and
rationale. Provide links to external references/discussions if appropriate.
If this pull request resolves any GitHub issues, link them like so:

  Resolves <link to issue>, resolves <link to another issue>.

For more information about linking a pull request to an issue, see:
https://docs.github.com/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/main/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

xedin committed:
[stdlib] Adopt `@_inheritActorContext(always)` on `Task.immediate`

This change aligns implementation with the current SE-0472 proposal.

xedin commented:
@swift-ci please test

ktoso reviewed (approved):
Lgtm, we may want to make the function always emit into client actually!

xedin commented:
@swift-ci please test Linux platform

xedin commented:
@swift-ci please test Linux platform

ktoso commented on stdlib/public/Concurrency/Task+immediate.swift.gyb:
```diff
@@ -72,7 +72,7 @@ extension Task where Failure == ${FAILURE_TYPE} {
     % # NOTE: This closure cannot be 'sending' because we'll trigger ' pattern that the region based isolation checker does not understand how to check'
     % #       In this case: `func syncOnMyGlobalActor() { Task.immediate { @MyGlobalActor in } }`
-    @_implicitSelfCapture _ operation: __owned @isolated(any) @escaping () async throws -> Success
+    @_inheritActorContext(always) operation: sending @isolated(any) @escaping () async throws -> Success
```
I think we do want the @_implicitSelfCapture so it feels similar to Task.init 🤔

ktoso commented on stdlib/public/Concurrency/Task+immediate.swift.gyb:
```diff
@@ -72,7 +72,7 @@ extension Task where Failure == ${FAILURE_TYPE} {
     % # NOTE: This closure cannot be 'sending' because we'll trigger ' pattern that the region based isolation checker does not understand how to check'
     % #       In this case: `func syncOnMyGlobalActor() { Task.immediate { @MyGlobalActor in } }`
-    @_implicitSelfCapture _ operation: __owned @isolated(any) @escaping () async throws -> Success
+    @_inheritActorContext(always) operation: sending @isolated(any) @escaping () async throws -> Success
```
Can you please mark those as `@_alwaysEmitIntoClient`? I should have done that earlier already.

xedin replied:
Sure!

xedin replied:
✅

xedin committed:
[stdlib] Concurrency: Add `@_alwaysEmitIntoClient` to add `Task.*immediate*`

xedin committed:
[stdlib] `Task.immediate` annotate `operation` parameter with `@_implicitSelfCapture`

Since the "operation" inherits the context of the actor it should
also be possible to reference its members without explicit use of
`self.`.

xedin commented:
@swift-ci please test

ktoso commented on test/api-digester/stability-concurrency-abi.test:
```diff
@@ -128,6 +128,10 @@ Func withThrowingTaskGroup(of:returning:body:) has parameter 2 type change from
 
+// Task.immediate APIs
+// _startTaskImmediately is `internal` and `@usableFromInline` because it has to be used by `Task.immediate` APIs that are all `@_alwaysEmitIntoClient`
+Func _startTaskImmediately(_:targetExecutor:) is a new API without '@available'
```
I guess instead we could give it 6.2 availability?

xedin replied:
✅

xedin commented:
@swift-ci please test

ktoso reviewed (approved):
Thanks, perfect now, lgtm
